tests/integration_tests/tests/connect_info.rs

```diff
@@ -48,3 +48,80 @@ async fn getting_connect_info() {
 
     jh.await.unwrap();
 }
+
+#[cfg(unix)]
+pub mod unix {
+    use std::path::Path;
+
+    use futures_util::FutureExt;
+    use tokio::{
+        net::{UnixListener, UnixStream},
+        sync::oneshot,
+    };
+    use tokio_stream::wrappers::UnixListenerStream;
+    use tonic::{
+        transport::{server::UdsConnectInfo, Endpoint, Server, Uri},
+        Request, Response, Status,
+    };
+    use tower::service_fn;
+
+    use integration_tests::pb::{test_client, test_server, Input, Output};
+
+    struct Svc {}
+
+    #[tonic::async_trait]
+    impl test_server::Test for Svc {
+        async fn unary_call(&self, req: Request<Input>) -> Result<Response<Output>, Status> {
+            let conn_info = req.extensions().get::<UdsConnectInfo>().unwrap();
+
+            // Client-side unix sockets are unnamed.
+            assert!(req.remote_addr().is_none());
+            assert!(conn_info.peer_addr.as_ref().unwrap().is_unnamed());
+            // This should contain process credentials for the client socket.
+            assert!(conn_info.peer_cred.as_ref().is_some());
+
+            Ok(Response::new(Output {}))
+        }
+    }
+
+    #[tokio::test]
+    async fn getting_connect_info() {
+        let unix_socket_path = "/tmp/uds-integration-test";
+
+        // tokio's `UnixListener` does not cleanup the socket automatically - we need to manually
+        // remove the file at the end of the test.
+        tokio::fs::create_dir_all(Path::new(unix_socket_path).parent().unwrap())
+            .await
+            .unwrap();
```

Comment:
nit: this just creates `/tmp`, are there situations where `/tmp` doesn't exist and we need to create it?

Reply:
Probably not? I think I'll just remove it, seems safe to assume we got `/tmp`